### ISSUE TITLE
ZIndex Xaml Markup Causes Crash on Release Build but Not Debug

### ISSUE THREAD

**Issue by speschka** [dotnet/maui#9554]
### Description

I have an app that is done. I've been testing it throughout by deploying the Debug builds to emulators and physical devices.

Today I tried deploying the Release build and the app won't run.  It just starts up and immediately stops.  I was fortunately able to capture some information from the Device Log, that included these entries:

Error	5903	AndroidRuntime	android.runtime.JavaProxyThrowable: System.FieldAccessException: Field `Microsoft.Maui.Controls.VisualElement:ZIndexProperty' is inaccessible from method `WatchLikesApp.MainPage:InitializeComponent ()'

Shortly after that was this:
Samsung SM-G998U	Debug	5903	AndroidRuntime	Shutting down VM
Samsung SM-G998U	Error	5903	AndroidRuntime	
at crc640ec207abc449b2ca.ShellSectionRenderer.n_onCreateView(Native Method)				
at crc640ec207abc449b2ca.ShellSectionRenderer.onCreateView(ShellSectionRenderer.java:42)				
at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2995)				
at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:523)				
at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)				
at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1840)				
at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1764)				
at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1701)				
at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2849)				
at androidx.fragment.app.FragmentManager.dispatchViewCreated(FragmentManager.java:2777)				
at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:3020)				
at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:551)				
at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)				
at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1840)				
at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1764)				
at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1701)				
at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2849)				
at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2784)				
at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:262)				
at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:478)				
at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:246)				
at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1455)				
at android.app.Activity.performStart(Activity.java:8315)				
at android.app.ActivityThread.handleStartActivity(ActivityThread.java:4060)				
at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)				
at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)				
at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)				
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)				
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)				
at android.os.Handler.dispatchMessage(Handler.java:106)				
at android.os.Looper.loopOnce(Looper.java:226)				
at android.os.Looper.loop(Looper.java:313)				
at android.app.ActivityThread.main(ActivityThread.java:8669)				
at java.lang.reflect.Method.invoke(Native Method)				
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)				
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)			

Again - this code works fine when deploying a Debug field, but crashes with a Release build.  To validate that the ZIndex property was the issue, I found the one control on the start up form where I set the ZIndex property in the Xaml.  I deleted the ZIndex property, deployed the Release build to my physical device again, and this time it worked.  

I added the ZIndex property back in, deployed the Release build to the same physical device, and it crashed again.

### Steps to Reproduce

I think the steps are covered in the problem description above.

### Version with bug

6.0.400

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android API 31

### Did you find any workaround?

Remove the ZIndex property, which I then was able to set in the OnNavigatedTo event in the code behind and it did not cause it to crash.

### Relevant log output

```shell
Included in the description above.
```

**Comment by NonameMissingNo** [comment]
Duplicate of #7331

**Comment by mfe-** [comment]
I'm running into this issue as well. Deployed the app always on Debug now with Release mode I got:

```logcat
08-22 19:54:21.301  1277  1277 E AndroidRuntime: FATAL EXCEPTION: main
08-22 19:54:21.301  1277  1277 E AndroidRuntime: Process: get.the.solution.t.App.DadPan, PID: 1277
08-22 19:54:21.301  1277  1277 E AndroidRuntime: android.runtime.JavaProxyThrowable: System.FieldAccessException: Field `Microsoft.Maui.Controls.VisualElement:ZIndexProperty' is inaccessible from method `t.App.View.GamePage:InitializeComponent ()'
08-22 19:54:21.301  1277  1277 E AndroidRuntime:
08-22 19:54:21.301  1277  1277 E AndroidRuntime:    at t.App.View.GamePage..ctor()
```

**Comment by jfversluis** [comment]
Thanks for the report! This is fixed in #8184 unfortunately it won't be released until .NET 7 because an API change was needed for this. Unfortunately there is not much we can do about it today.

**Comment by mfe-** [comment]
@jfversluis 
I use `Zindex` as overlay to display an animation over the whole current page. Is there a workaround for this?
